src/ssh/scp_client.cpp

```diff
@@ -52,7 +52,17 @@ mp::SCPClient::SCPClient(SSHSessionUPtr ssh_session) : ssh_session{std::move(ssh
 
 void mp::SCPClient::push_file(const std::string& source_path, const std::string& destination_path)
 {
-    SCPUPtr scp{make_scp_session(*ssh_session, SSH_SCP_WRITE, destination_path.c_str())};
+    auto full_destination_path{destination_path};
+    if (full_destination_path.empty())
+    {
+        full_destination_path.append(QFileInfo(QString::fromStdString(source_path)).fileName().toStdString());
+    }
+    else if (QFileInfo(QString::fromStdString(destination_path)).isDir())
+    {
+        full_destination_path.append("/" + QFileInfo(QString::fromStdString(source_path)).fileName().toStdString());
+    }
```

Comment:
Perhaps the above can be broken into:

A couple of utility functions (maybe even include them in utils.h?)
```
auto filename_for(const std::string& path)
{
    return QFileInfo(QString::fromStdString(path)).fileName().toStdString();
}

bool is_dir(const std::string& path)
{
    return QFileInfo(QString::fromStdString(path)).isDir();
}
```

Then you can lambda initialize full_destination_path like:

```
const std::string full_destination_path = [&]{
        if (destination_path.empty())
            return filename_for(source_path);
        if (is_dir(destination_path))
            return destination_path + "/" + filename_for(source_path);
        return destination_path;
    }();

```

Though I see this is the same logic as in pull_file, maybe it should be a full function, like:

```
std::string full_destination(const std::string& source, const std::string& dest)
{
...
}
```